tests/org.eclipse.xtext.builder.tests/src/org/eclipse/xtext/builder/builderState/EclipseResourceFileSystemAccess2Test.java

```diff
@@ -28,23 +29,28 @@
  * @author Moritz Eysholdt - Initial contribution and API
  */
 public class EclipseResourceFileSystemAccess2Test extends Assert {
+	private IProject project;
+	private EclipseResourceFileSystemAccess2 fsa;
 
-	@After
-	public void tearDown() throws Exception {
-		IResourcesSetupUtil.cleanWorkspace();
-	}
-
-	@Test
-	public void testWriteReadFiles() throws Exception {
-		IProject project = IResourcesSetupUtil.createProject("test");
+	@Before
+	public void setUp () throws Exception {
+		project = IResourcesSetupUtil.createProject("test");
 		Injector injector = Activator.getInstance().getInjector(
 				Activator.ORG_ECLIPSE_XTEXT_BUILDER_TESTS_BUILDERTESTLANGUAGE);
-		EclipseResourceFileSystemAccess2 fsa = injector.getInstance(EclipseResourceFileSystemAccess2.class);
+		fsa = injector.getInstance(EclipseResourceFileSystemAccess2.class);
 		fsa.setProject(project);
 		fsa.setOutputPath("src-gen");
 		fsa.getOutputConfigurations().get(IFileSystemAccess.DEFAULT_OUTPUT).setCreateOutputDirectory(true);
 		fsa.setMonitor(new NullProgressMonitor());
+	}
 
+	@After
+	public void tearDown() throws Exception {
+		IResourcesSetupUtil.cleanWorkspace();
```

Comment:
Minor: Some test classes share a common test project to speed up test execution (not in this particular test suite, though). Recently we tried to avoid a full clean of the workspace and rather remove the projects that a test class created.